base/tps/src/org/dogtagpki/server/tps/channel/SecureChannel.java

```diff
@@ -1165,7 +1158,12 @@ public TPSBuffer readObject(TPSBuffer objectID, int offset, int len) throws TPSE
 
         while (sum < len) {
 
+CMS.debug("sum: " + sum + " cur_offset: " + cur_offset + " cur_read: " + cur_read);
             read = new ReadObjectAPDU(objectID.toBytesArray(), cur_offset, cur_read);
+            // Add a 0x00 Le byte
+            read.setTrailer(new TPSBuffer((byte) 0x00));
+
+CMS.debug("read encoding: " + read.getEncoding().toHexString());
```

Comment:
might want to comment this debug out.

Reply:
Good catch, that was for debugging the token side keygen case for one of the tokens.